adapters/solana/da_client/src/main.rs

```diff
@@ -0,0 +1,361 @@
+use anchor_client::{Client, Cluster};
+use solana_sdk::commitment_config::{CommitmentConfig, CommitmentLevel};
+use solana_sdk::signature::EncodableKey;
+use solana_sdk::{
+    pubkey::Pubkey,
+    signature::{Keypair, Signature, Signer},
+    signer::keypair::read_keypair_file,
+    system_instruction, system_program,
+    transaction::Transaction,
+};
+use std::path::Path;
+use std::str::FromStr;
+
+use anchor_lang::solana_program::sysvar::clock::Clock;
+use solana_sdk::sysvar::SysvarId;
+
+use blockroot::accounts as blockroot_accounts;
+use blockroot::instruction as blockroot_instruction;
+use blockroot::{get_chunks, Chunk, ChunkAccumulator, CHUNK_SIZE, PREFIX};
+
+use solana_rpc_client::rpc_client::RpcClient;
+
+use clap::{Parser, Subcommand};
+use std::process;
+
+use alloc::rc::Rc;
+extern crate alloc;
+
+use da_client::{read_file_to_vec, write_random_bytes};
+
+const DEFAULT_RPC_URL: &str = "http://localhost:8899";
+const DEFAULT_WS_URL: &str = "ws://localhost:8900";
+
+fn create_account(
+    url: &str,
```

Comment:
These arguments are common to the functions in this file. 
```
url: &str,
payer: &Keypair,
blockroot_program: &str,
chunks_account: &str,
```
Does it make sense to extract it to a separate structure and implement `create_account ` as methods of that struct (so it would take &self)?

Reply:
yup that makes sense!